daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -268,6 +317,10 @@ abstract class BinaryDecimalUnparserBase(
     nBits: Int,
     finfo: FormatInfo
   ): Boolean = {
-    dos.putBigInt(asBigInt(value), nBits, signed == YesNo.Yes, finfo)
+    if (nBits > 0) {
+      dos.putBigInt(asBigInt(value), nBits, signed == YesNo.Yes, finfo)
+    } else {
+      true
```

Comment:
Should this be `false`? I think the purpose of the bug is to make it change it so zero-length integers are now an error? If you put the nBits == 0 check in the `unparse` method we can handle this case for all binary numbers.

Reply:
I think the reason I moved the check out of the unparse was we wouldn't be able to reach the code in BinaryIntegerBaseUnparser.puNumber in the case that nBits was 0

Reply:
I thought we never need to unparse 0 bits? Isn't unparsing 0 bits always illegal now?